library/alloc/src/boxed.rs

```diff
@@ -1431,3 +1432,16 @@ impl<F: ?Sized + Future + Unpin, A: AllocRef> Future for Box<F, A> {
         F::poll(Pin::new(&mut *self), cx)
     }
 }
+
+#[unstable(feature = "async_stream", issue = "79024")]
+impl<S: ?Sized + Stream + Unpin> Stream for Box<S> {
+    type Item = S::Item;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        Pin::new(&mut **self).poll_next(cx)
```

Comment:
why the 2 *'s here? the first DerefMut's the Pin, which actually deref's the box: https://doc.rust-lang.org/src/core/pin.rs.html#785-789, the second is extraneous I believe (https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=528d5daa795f4c38210697a6919e67f1), should we match how the future parallel does this: https://doc.rust-lang.org/src/alloc/boxed.rs.html#1162-1168?

I know this is a tiny detail, but I want to make sure I understand if the 2 *'s are necessary

Reply:
`Pin::new(&mut **self)` returns `Pin<&mut S>`, `Pin::new(&mut *self)` returns `Pin<&mut Box<S>>` (same type as `self`).
I'm not sure why `Future`'s one is actually work, but 2 * should be more correct. (actually, `impl Future for &mut F` uses that.)

Reply:
ohhhh self is a `Pin<&mut Box<S>` so the first * deref's the &mut. autoderef might be doing something funky but you are right, 2 *'s is more explicit and better here